### PR TITLE
Added fsh block to toolbox and changed how it behaves

### PR DESCRIPTION
**Pull request by inventionpro**
added inv_fsh_api_animal block with it's own category under applications
and changed it's code a little

### CODE REVIEW

**Comment by JeremyGamer13** [comment]
recommendations:
adding the list of available animals in the block's tooltip as it would be useful to know
removing the 2 logs and replace the error log to use console.error (console.log is fine but console.error makes more sense)

**Comment by inventionpro** [comment]
> recommendations: adding the list of available animals in the block's tooltip as it would be useful to know removing the 2 logs and replace the error log to use console.error (console.log is fine but console.error makes more sense)

list of available animals is in blocks help url and logs stuff was from frozer but thanks for recommendations will probably add them

**Review by Malix-Labs — CHANGES_REQUESTED** [review]
S4D's private variables and functions should start with `_S4D_`

**Comment by inventionpro** [comment]
> S4D's private variables and functions should start with `_S4D_`

done

**Comment by Malix-Labs** [comment]
@JeremyGamer13:
> recommendations:
> adding the list of available animals in the block's tooltip as it would be useful to know
> removing the 2 logs and replace the error log to use console.error (console.log is fine but console.error makes more sense)

Pls review if this recommendation has been implemented

**Comment by inventionpro** [comment]
@JeremyGamer13

**Comment by inventionpro** [comment]
> @JeremyGamer13:
> 
> > recommendations:
> > adding the list of available animals in the block's tooltip as it would be useful to know
> > removing the 2 logs and replace the error log to use console.error (console.log is fine but console.error makes more sense)
> 
> Pls review if this recommendation has been implemented

it has?

**Comment by JeremyGamer13** [comment]
> > @JeremyGamer13:
> > > recommendations:
> > > adding the list of available animals in the block's tooltip as it would be useful to know
> > > removing the 2 logs and replace the error log to use console.error (console.log is fine but console.error makes more sense)
> > 
> > 
> > Pls review if this recommendation has been implemented
> 
> it has?

it seems ye
i think malix needs to review the other thing now

**Comment by inventionpro** [comment]
how much more time do i need to wait?

**Comment by RedMan13** [comment]
idfk normally its merged by now

**Review by Malix-Labs — APPROVED** [review]

**Review by Malix-Labs — COMMENTED** [review]
Is the toolbox category which in those blocks are merged okay?

File: src/toolbox.js [hunk 1]
```diff
@@ -2797,6 +2797,11 @@ export default (favorites) => {
             <block type="generate_image_openai"/>
             <block type="text_completion_openai" />
         </category>
+        <category name="Fsh API" colour="#50494e">
+            <label text="Fsh api"/>
+            <label text="Image api"/>
+            <block type="inv_fsh_api_animal"/>
+        </category>
```
Comment (Malix-Labs) [inline src/toolbox.js]: Is this toolbox location okay for you @scratch-for-discord/toolbox?
Maybe it would be better more at the bottom, no?

Comment (inventionpro) [reply src/toolbox.js:2804]: ye?

Comment (inventionpro) [reply src/toolbox.js]: no

Comment (Malix-Labs) [reply src/toolbox.js]: > ye?

That's for @scratch-for-discord/toolbox to answer.

Your edit works, that's not of debate;
the debate is "is it where it should be?"

**Comment by inventionpro** [comment]
is this ever getting merged?

File: src/toolbox.js (hunk 1 above)
Comment (JeremyGamer13) [reply src/toolbox.js]: at the bottom of what because its at the bottom of applications already

Comment (Malix-Labs) [reply src/toolbox.js]: > at the bottom of what because its at the bottom of applications already

Ye, didn't noticed.

**Comment by inventionpro** [comment]
@Malix-Off mind already accepting pull request

**Comment by Malix-Labs** [comment]
> @Malix-Off mind already accepting pull request

[It's on you now](https://github.com/scratch-for-discord/Web-Application_Frontend/pull/201#event-8724637434)

**Pull request merged**